### PR TITLE
[FEATURE] Tableau Hyper

### PR DESCRIPTION
**Pull request by maxim-mityutko**
Provide the capability to write into the Tableau Hyper format, that allows loading datasource data independently from the Tableau server.

## Description

Writing data into hyper from List, Parquet, Dataframe.
Reading data from Hyper into Dataframe.
Publishing Hyper to the Tableau server.

## Related Issue

#45 

## Motivation and Context

In certain scenarious loading data into the datasource by means of Tableau server may take too much time (and depending on Tableau server timeouts, may even fail). Creating Hyper files independently from Tableau allows much faster load times.

## How Has This Been Tested?

Unit tests / Publishing to the Tableau server

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/koheesio/integrations/spark/tableau/hyper.py [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    @abstractmethod
+    def execute(self):
+        pass
+
+
+class HyperFileListWriter(HyperFileWriter):
+    """
+    Write list of rows to a Hyper file.
+
+    Reference:
```
Comment (dannymeijer) [inline src/koheesio/integrations/spark/tableau/hyper.py:179]: Please reformat this docstring to be Numpy compliant.

- Change headings to be like this
    > Reference
    > ----------
- Ensure the Example heading is plural (Examples)
- Put the code example in a code-block

File: src/koheesio/integrations/spark/tableau/hyper.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+                with Inserter(connection, self.table_definition) as inserter:
+                    inserter.add_rows(rows=self.data)
+                    inserter.execute()
+
+        self.output.hyper_path = self.hyper_path
+
+
+class HyperFileParquetWriter(HyperFileWriter):
+    """
+    Read one or multiple parquet files and write them to a Hyper file.
```
Comment (dannymeijer) [inline src/koheesio/integrations/spark/tableau/hyper.py:222]: same as above

File: src/koheesio/integrations/spark/tableau/server.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    TableauAuth,
+)
+
+from pydantic import Field, SecretStr
+
+from koheesio.models import model_validator
+from koheesio.steps import Step, StepOutput
+
+
+class TableauServer(Step):
```
Comment (dannymeijer) [inline src/koheesio/integrations/spark/tableau/server.py:22]: Please add docstring

File: src/koheesio/integrations/spark/tableau/server.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+class TableauHyperPublishMode(str, Enum):
+    """
+    The different publishing modes for the TableauHyperPublisher.
+    """
+
+    APPEND = Server.PublishMode.Append
+    OVERWRITE = Server.PublishMode.Overwrite
+
+
+class TableauHyperPublisher(TableauServer):
```
Comment (dannymeijer) [inline src/koheesio/integrations/spark/tableau/server.py:178]: Please add docstring

**Review by dannymeijer — CHANGES_REQUESTED** [review]
I like the code. Please have a look at the docstrings though. Would like to see every class have a docstring, and ensure docstring formats are Numpy style.

File: src/koheesio/integrations/spark/tableau/hyper.py [hunk 5]
```diff
@@ -270,6 +284,22 @@ def execute(self):
...
+    Examples
+    --------
+    ```python
+    hw = HyperFileDataFrameWriter(
+        df=spark.createDataFrame([(1, "foo"), (2, "bar")], ["id", "name"]),
+        name="test",
+    ).execute()
+
+        # do somthing with returned file path
+        hw.hyper_path
```
Comment (dannymeijer) [inline src/koheesio/integrations/spark/tableau/hyper.py:301]: missing the closing `````

Comment (maxim-mityutko) [reply src/koheesio/integrations/spark/tableau/hyper.py]: added it, thanks for spotting!

**Review by dannymeijer — APPROVED** [review]

**Pull request merged**